sky/shell/shell.cc

```diff
@@ -85,12 +165,23 @@ Shell::Shell() {
   ui_thread_.reset(new base::Thread("ui_thread"));
   ui_thread_->StartWithOptions(options);
   ui_task_runner_ = ui_thread_->message_loop()->task_runner();
+  ui_task_runner_->PostTask(
+      FROM_HERE, base::Bind(&Shell::InitUIThread, base::Unretained(this)));
 
   io_thread_.reset(new base::Thread("io_thread"));
   io_thread_->StartWithOptions(options);
   io_task_runner_ = io_thread_->message_loop()->task_runner();
 
   blink::SetServiceIsolateHook(ServiceIsolateHook);
+
+  Dart_RegisterRootServiceRequestCallback(
+      "_flutter.runInView",
+      &FlutterRunInView,
+      nullptr);
+  Dart_RegisterRootServiceRequestCallback(
+      "_flutter.listViews",
+      &FlutterListViews,
+      nullptr);
```

Comment:
I'd move these calls out of this file as well and have a call similar to `blink::SetServiceIsolateHook` for this layer of the system.